pages/collection/[chainId]/[id].tsx

```
@@ -83,14 +82,14 @@ export const getServerSideProps = wrapServerSideProps<Props>(
         ? ctx.params.chainId[0]
         : ctx.params.chainId
       : null
-    invariant(chainIdStr, 'chainId is required')
+    if (!chainIdStr) return { notFound: true }
     const chainId = parseInt(chainIdStr, 10)
     const collectionAddress = ctx.params?.id
       ? Array.isArray(ctx.params.id)
         ? ctx.params.id[0]
         : ctx.params.id
       : null
-    invariant(collectionAddress, 'Collection Address is required')
+    if (!collectionAddress) return { notFound: true }
```

I left a comment on another PR where this is not entirely true, and in this case, we should throw an error as this means that next is not doing the routing properly.

This check should never occur as we don't validate the data we just retrieve it.

The flow should be:
- Retrieve the data from the params
- If they are supposed to be in the URL, and they don't exist, then throw an error (as next is broken somewhere)
- Process the data (doing a parseInt, for example)
- If there is an error, then ideally, we should throw a 400 (or ignore it and go to the next step)
- Check the presence of the resource (query the API)
- If the resource doesn't exist (because data mal formatted or the resource is inexistent), then throw a 404

That said, this might be a bit too much/overkill, and I'm ok to return some 404 in that case, even if that's not really correct, but I just wanted to highlight this.

If we could keep the 404 only at the last step, that would be great tho :)
Doing that will, I believe reduce drastically the size of this PR

---

Let me know if I get the idea correctly with the most recent commit please 😄